### DISCUSSION TITLE
Dataset.plot

### DISCUSSION THREAD

**Discussion by headtr1ck** [pydata/xarray#6604] [Ideas]
We often use the workflow of using Datasets that contain the same "datatype" of different sensors (i.e. two cameras with different resolution that capture the same object).

A functionality to simply call `ds.plot()` would simplify the workflow alot.

This plot should create a grid and plot each variable in a subfigure.

Optional args/kwargs:
- variables: Sequence[Hashable] which variables to plot, or the order
- nrows
- ncols

This simple idea would only work for 1D and 2D data.
One could go even further and add the `col` argument to plot a series of plots for each variable and thus support 3D data.

**Comment by max-sixty** [comment]
I agree this would be useful! There's some challenge in finding an approach that works universally. To the extent that each dataset is dimensioned the same, a grid seems good. But there will also be different cases.

Pandas has an approach with `df.plot.hist(...)`, i.e. a namespace within `plot`.

What do others think about adding a `ds.plot()` with subplots?

**Comment by OriolAbril** [comment]
I am aiming to get to something similar (or at least the infrastructure to easily get something like this going). It is still very early stage, but feedback is welcome:

* Initial idea with DataArrays but more flexible than current FacetGrid when it comes to aesthetics mapping: https://xrtist.readthedocs.io/en/latest/data_organizers.html
* Adaptation to Dataset: https://xrtist.readthedocs.io/en/latest/plot_museum.html

**Comment by TomNicholas** [comment]
This crosses over a lot with the [same issue in Datatree](https://github.com/xarray-contrib/datatree/issues/47). Fundamentally users commonly fill the data structure with a simpler and easier to reason about set of data than what we developers worry it might be containing.

I also agree with @max-sixty that you don't want to be too hasty to commandeer the top-level .plot() name.
